src/m365/spo/commands/file/file-sharinginfo-get.ts

```diff
@@ -0,0 +1,171 @@
+
+  public get name(): string {
+    return commands.FILE_SHARINGINFO_GET;
+  }
+
+  public get description(): string {
+    return 'Generates the sharing information report for specified file';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.id = (!(!args.options.id)).toString();
+    telemetryProps.url = (!(!args.options.url)).toString();
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    if (this.verbose) {
+      logger.log(`Retrieving sharing information report for the file...`);
+    }
+
+    this.getneededFileInformation(args, logger)
+      .then((fileInformation: { fileItemId: number; libraryName: string; }): Promise<string> => {
+        if (this.verbose) {
+          logger.log(`Retrieving Sharing information report for the file with item Id  ${fileInformation.fileItemId}`);
+        }
+
+        const requestOptions: any = {
+          url: `${args.options.webUrl}/_api/web/lists/getbytitle('${fileInformation.libraryName}')/items(${fileInformation.fileItemId})/GetSharingInformation?$select=permissionsInformation&$Expand=permissionsInformation`,
+          headers: {
+            'accept': 'application/json;odata=nometadata'
+          },
+          responseType: 'json'
+        };
+        return request.post(requestOptions)
+      }).then((res: any): void => {
+        if (args.options.output === 'json') {
+          logger.log(res);
+        }
+        else {
+          let FileSharingInfoCollection: any[] = [];
+
+          res.permissionsInformation.links.map((links: any) => {
+            links.linkDetails.Invitations.map((linkInvites: any) => {
+              const FileSharingInfo: any = {
+                SharedWith: linkInvites.invitee.name,
+                IsActive: linkInvites.invitee.isActive,
+                IsExternal: linkInvites.invitee.isExternal,
+                PrincipalType: FileSharingPrincipalType[parseInt(linkInvites.invitee.principalType)]
+              };
+              FileSharingInfoCollection.push(FileSharingInfo);
+            })
+          })
+          res.permissionsInformation.principals.map((principals: any) => {
+            const FileSharingInfo: any = {
+              SharedWith: principals.principal.name,
+              IsActive: principals.principal.isActive,
+              IsExternal: principals.principal.isExternal,
+              PrincipalType: FileSharingPrincipalType[parseInt(principals.principal.principalType)]
+            };
+            FileSharingInfoCollection.push(FileSharingInfo);
+          });
+
+          logger.log(FileSharingInfoCollection.map((r: any) => {
+            return {
+              SharedWith: r.SharedWith,
+              IsActive: r.IsActive,
+              IsExternal: r.IsExternal,
+              PrincipalType: r.PrincipalType,
+            }
+          }));
+        }
```

Comment:
Hello @waldekmastykarz  - Based on [THIS Conversation](https://github.com/pnp/cli-microsoft365/issues/1659#issuecomment-722377082), apart from the normal review, these are the lines which I am doubtful about. I am not very sure the approach which I have followed is correct or not. 

Just giving some more information.

- The result information are available in multiple object arrays.
- Here I am merging both the values into a single object and is returning the same.
- Can you please review the same and let me know the approach is correct or not.

Thanks again.🙏

Reply:
Just to double check: the flattened array would be for text output only, right? In JSON output, we'd output the data as retrieved from the API?

Reply:
That is correct @waldekmastykarz . [We had decided](https://github.com/pnp/cli-microsoft365/issues/1659#issuecomment-722377082) that we will not be altering the JSON result since users may built scripts which will use the data as per their business process

Reply:
I'd say, let's give it a try. The only change I'd suggest at this point is to rename `FileSharingInfo` and `FileSharingInfoCollection` to start with lowercase `f` but other than that, let's give it a shot, see how it works and adjust if necessary.

Reply:
Thanks a lot for the guidance on that. I will be changing the status **Ready for Review** so rest of the command could be reviewed.

Thanks again for spending your time on that.. ♥️